sbt-version-policy/src/main/scala/sbtversionpolicy/SbtVersionPolicySettings.scala

```diff
@@ -160,6 +161,7 @@ object SbtVersionPolicySettings {
             scalaModuleInf,
             updateConfig,
             warningConfig,
+            extractVersion,
```

Comment:
There is something subtle going on here: we pass a `PartialFunction` where a (total) function is expected. This is not safe (for instance, try with `versionPolicyModuleVersionExtractor := { case "dummy" => "0.0.0" }` in the tests).

I realize my initial suggestion (using `orElse` on the partial function) does not work as I thought because it returns back a partial function again. I would argue that we miss a method `fallbackTo` on `PartialFunction` but that’s another debate. We have to work with `applyOrElse` instead, which address exactly our needs.

I suggest the following course of action:

- Set the default value of `versionPolicyModuleVersionExtractor` to `PartialFunction.empty`
- Change the signature of `DependencyCheckmodulesOf` and `DependencyCheck.report` to take a `PartialFunction` instead of a function
- Directly pass `versionPolicyModuleVersionExtractor.value` to them
- When you apply it, call `moduleToVersion.applyOrElse(mod, _.revision)` (you should probably put that expression to a utility method)

What do you think?